### PR TITLE
Update monitoring branch and Fix Overview screenshot

### PR DESCRIPTION
**Pull request by k0machi**
- test(test_default.yaml): set monitoring version to 4.5
- fix(monitorstack/ui.py): Fix panel name for 4.4/4.5 monitoring

Closes #6408 

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by k0machi** [comment]
There's another issue that I found is that the "Publish snapshot to raintank.io" feature seems to be missing, but I'm not sure if that's being used yet, so I haven't fixed it yet.

**Comment by k0machi** [comment]
[Argus](https://argus.scylladb.com/test/b0b22734-758e-47b4-b6ec-29e64d212d52/runs?additionalRuns[]=a9f0932b-a8da-4bd5-96bc-f2788f226061)

**Comment by fruch** [comment]
> [Argus](https://argus.scylladb.com/test/b0b22734-758e-47b4-b6ec-29e64d212d52/runs?additionalRuns%5B%5D=a9f0932b-a8da-4bd5-96bc-f2788f226061)

those jobs still show only one screenshot...

**Review by fruch — CHANGES_REQUESTED** [review]
still missing one of the screenshots

**Comment by fruch** [comment]
> > [Argus](https://argus.scylladb.com/test/b0b22734-758e-47b4-b6ec-29e64d212d52/runs?additionalRuns%5B%5D=a9f0932b-a8da-4bd5-96bc-f2788f226061)
> 
> those jobs still show only one screenshot...

@k0machi also you don't need such long job to get those screenshots, you can run the docker provision test (or aws/gce ones on this PR)

**Comment by k0machi** [comment]
Okay, I've corrected the locators for real this time, there seems to be an issue with screenshots specifically for argus, looking into that now, as monitor-set.tar.gz contains both screenshots

**Comment by k0machi** [comment]
And after further investigation it appears that screenshots are fixed as long as there are no failures during the run or if the test takes too long - You can see that `#57` in Argus has both screenshots without any changes, it appears that there's an issue with Docker Container object, because overview screenshot fails instantly with "Cannot get screenshot" when trying to call `self.browser.set_window_size`

**Comment by k0machi** [comment]
Provision test seems to be passing with both screenshots, running my staging job to check if adding a retry for docker reload fixes the issue.

**Comment by soyacz** [comment]
> Okay, I've corrected the locators for real this time, there seems to be an issue with screenshots specifically for argus, looking into that now, as monitor-set.tar.gz contains both screenshots.
> 
> UPD: I can see that email however is also missing a second screenshot, so problem is somewhere between email_data and collecting them

IIRC we collect screenshots 2 times - in teardown and in 'collect-logs' stage. So looks that it works in teardown and breaks the next collection. Maybe we don't close properly browsers (or other similar problem)?

**Comment by k0machi** [comment]
> IIRC we collect screenshots 2 times - in teardown and in 'collect-logs' stage. So looks that it works in teardown and breaks the next collection. Maybe we don't close properly browsers (or other similar problem)?

To me it seems that docker daemon stops working for some reason, maybe something shuts it off? One thing is that it's only the overview screenshot that fails - eventually when the collector moves to the per-nemesis screenshot, the daemon suddenly responds and works. This seems to not happen if a longevity is short - I've shortened `#57` to one hour and both screenshots got collected fine. What I can see is that overview screenshot seems to be collected fairly often, not just in teardown and collect logs stage, there are at least 5-6 spots where it's collected successfuly (during nemeses I think).

File: sdcm/utils/docker_utils.py [hunk 1]
```diff
@@ -176,15 +176,23 @@ def _get_attr_for_name(instance: object,
...
         container = instance._containers.get(name)
         if container is None and raise_not_found_exc:
             raise NotFound(f"There is no container `{name}' for {instance}.")
         if container:
-            container.reload()
+            try:
+                container.reload()
+            except OSError:
+                LOGGER.warning("Container failed to refresh, checking docker connection...")
+                docker_client = cls.get_docker_client(instance, name)
```
Comment (soyacz) [inline sdcm/utils/docker_utils.py:191]: `cls.get_docker_client` if it gets `name` param it runs `cls.get_container` - so we have recurrence here :)
I propose to not provide name for `cls.get_docker_client` in that case.

Comment (k0machi) [reply sdcm/utils/docker_utils.py]: I've added code for that, let's see how this will behave now.

**Comment by k0machi** [comment]
Latest run contains both screenshots, however the exception handler didn't start in it, so may still not be fixed, but looks good so far.

**Comment by soyacz** [comment]
> Latest run contains both screenshots, however the exception handler didn't start in it, so the docker issue may still not be fixed, but looks good so far.

ok, so let's merge this one and handle docker issues (if they appear) in different PR.

**Review by soyacz — APPROVED** [review]

File: sdcm/utils/docker_utils.py [hunk 2]
```diff
@@ -78,6 +78,12 @@ def __bool__(self):
         return self.full is not None
 
 
+class INodeWithContainerManager(Protocol):
```
Comment (aleksbykov) [inline sdcm/utils/docker_utils.py:81]: Why this needed as Protocol?

**Review by aleksbykov — APPROVED** [review]
LGTM, short question

File: sdcm/utils/docker_utils.py (hunk 2 above)
Comment (k0machi) [reply sdcm/utils/docker_utils.py:81]: Protocol is here for static type checking, it's a class that describes an interface

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**